src/KodKod.Analyzer/ApiActionsAreAttributeRoutedFixProvider.cs

```diff
@@ -0,0 +1,112 @@
+        {
+            { "Get", HttpGetAttribute },
+            { "Find", HttpGetAttribute },
+            { "Post", HttpPostAttribute},
+            { "Put", HttpPutAttribute },
+            { "Delete", HttpDeleteAttribute },
+        };
+
+        public sealed override ImmutableArray<string> FixableDiagnosticIds =>
+            ImmutableArray.Create(DiagnosticDescriptors.KK1000_ApiActionsMustBeAttributeRouted.Id);
+
+        public sealed override FixAllProvider GetFixAllProvider()
+        {
+            return WellKnownFixAllProviders.BatchFixer;
+        }
+
+        public sealed override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var diagnostic = context.Diagnostics[0];
+            var methodName = diagnostic.Properties["MethodName"];
+            var idParameter = diagnostic.Properties["IdParameterName"];
+
+            foreach (var mapping in KeyWordMappings)
+            {
+                if (methodName.StartsWith(mapping.Key, StringComparison.Ordinal))
+                {
+                    var title = $"Add {mapping.Key} attribute";
+                    context.RegisterCodeFix(
+                        CodeAction.Create(
+                            title,
+                            createChangedDocument: cancellationToken => CreateChangedDocumentAsync(mapping.Value, cancellationToken),
+                            equivalenceKey: title),
+                        context.Diagnostics);
+
+                    return Task.CompletedTask;
+                }
+            }
+
+            foreach (var attribute in Attributes)
+            {
+                var title = $"Add {attribute.Key} attribute";
+                context.RegisterCodeFix(
+                    CodeAction.Create(
+                        title,
+                        createChangedDocument: cancellationToken => CreateChangedDocumentAsync(attribute.Value, cancellationToken),
+                        equivalenceKey: title),
+                    context.Diagnostics);
+            }
+
+            return Task.CompletedTask;
+
+            async Task<Document> CreateChangedDocumentAsync(string attributeName, CancellationToken cancellationToken)
+            {
+                var rootNode = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+                var methodNode = (MethodDeclarationSyntax)rootNode.FindNode(context.Span);
+
+                var editor = await DocumentEditor.CreateAsync(context.Document, cancellationToken).ConfigureAwait(false);
+
+                var attributeMetadata = editor.SemanticModel.Compilation.GetTypeByMetadataName(attributeName);
+                attributeName = attributeMetadata.ToMinimalDisplayString(editor.SemanticModel, methodNode.SpanStart);
+                attributeName = attributeName.Replace("Attribute", null);
+
+                var route = methodName;
```

Comment:
I think it's better to leave the method name out of the route by default

Reply:
Also, what about putting a route on the class?

I'm not 100% certain we know exactly the scenarios for fixing this. - IE what are the cases that existing, what can we detect and what should be done? We need to enumerate those things. This is a decent start.